x-pack/plugins/integration_assistant/server/util/samples.ts

```diff
@@ -233,3 +234,45 @@ export function mergeSamples(objects: any[]): string {
 
   return JSON.stringify(result, null, 2);
 }
+
+export function flattenObjectsList(
+  obj: Field[]
+): Array<{ name: string; type: string; description?: string }> {
+  const result = flattenObject(obj);
+
+  return sortListOfObjects(result);
+}
+
+function flattenObject(
+  obj: Field[],
+  parentKey: string = '',
+  separator: string = '.'
+): Array<{ name: string; type: string; description?: string }> {
+  let result: Array<{ name: string; type: string; description?: string }> = [];
+
+  obj.forEach((element) => {
+    if (element.name) {
+      const newKey = parentKey ? `${parentKey}${separator}${element.name}` : element.name;
+
+      if (element.fields && Array.isArray(element.fields)) {
+        result = result.concat(flattenObject(element.fields, newKey, separator));
```

Comment:
What we have here is the classical tree walking problem. Generally one can walk the tree in linear time.

The specific `result = result.concat` has quadratic scaling as it produces a bunch of arrays of the form 
```
[A]
[A, B]
[A, B, C]
...
```

There are some possible strategies for improvement:

1. Push the new results into `result` to go to `n log(n)` scaling (each level of the tree merges all results below)
2. Pass `result` into the function and collect data there without creating new storage (classic, but good).
3. Convert the function into a generator and make it yield recursively (I think it's the same as before, but has nicer syntax)
4. Walk the tree without recursion (something like [this](https://codingots.medium.com/tree-traversal-without-recursion-221cbea6d004))
5. Check if we have a generic tree walking function somewhere in Kibana or used third-party libraries.